src/java.desktop/share/classes/javax/swing/border/LineBorder.java

```diff
@@ -161,7 +163,8 @@ private void paintUnscaledBorder(Component c, Graphics g,
             Shape outer;
             Shape inner;
 
-            int offs = this.thickness * (int) scaleFactor;
+            int offs = clipRound(this.thickness*scaleFactor);
+
```

Comment:
I'd leave out the added blank line, both `offs` and `size` are still tightly coupled, only the formula for `offs` is amended.